cvat/apps/dataset_manager/annotation.py

```diff
@@ -950,6 +951,9 @@ def propagate(shape, end_frame, *, included_frames=None):
                 break # The track finishes here
 
             if prev_shape:
+                if curr_frame == prev_shape["frame"]:
+                    if dict(shape, id=None, keyframe=None) == dict(prev_shape, id=None, keyframe=None):
```

Comment:
These 2 `if` statements can be combined into one.
Why do we need to exclude `keyframe` when comparing shapes?

Reply:
They can be combined, but then this line becomes too long (137 cahracters) and should be split to 2 lines anyway, and I think it is more readable the way it is now. 
`keyframe` should be excluded, because it is always `True` in `prev_shape` due to `shape["keyframe"] = True` on line 967.

Reply:
> They can be combined, but then this line becomes too long (137 cahracters) and should be split to 2 lines anyway, and I think it is more readable the way it is now.
```python
if (
    condition1
    and condition2
):
    ...
```

Reply:
ok, combined them